src/components/node.js

```diff
@@ -37,6 +37,7 @@ class Node extends React.Component {
     parent: React.PropTypes.object.isRequired,
     readOnly: React.PropTypes.bool.isRequired,
     schema: React.PropTypes.object.isRequired,
+    selectionNeedsRedraw: React.PropTypes.bool.isRequired,
```

Comment:
Unfortunately we can't use `data.isNative`.
First of all, `updateSelection` is triggered by selection changes that are generated programmatically and not by native selection changes. Indeed when the changes are consequences of native selection transformations the two selections are already in sync and there is no need to call 'updateSelection'.
Moreover setting `data.isNative` to `true` stops the updating of the whole editor content, but when selection changes, some components need re-rendering: for example `Void` `Inline` nodes which want to show an highlight border if selected.
The solution I adopted is to split in two the `Leaf` component.
The *higher* component, still named `Leaf`, is responsible for updating the native selection, the *lower*, named `LeafContent`, instead is used to render the leaf text content.
In this way we can update the `Leaf`, and therefore the native selection, without touching its content.